src/Generator/GenerateCommand.php

```diff
@@ -0,0 +1,228 @@
+
+        $configPath = $this->getConfigFile($input);
+        $io->comment($configPath === null
+            ? 'No config file. Default configuration applied.'
+            : sprintf('Using config file "%s".', $configPath)
+        );
+
+        try {
+            $config = $configPath
+                ? $this->configFactory->createFromFile($configPath)
+                : $this->configFactory->createDefault();
+
+            $generator = $this->generatorFactory->create($config);
+            foreach ($sourceFileCollection->getExisting() as $sourceFile) {
+                $result = $generator->generate($sourceFile, $this->getOverrideFlag($input));
+                $io->success(sprintf('Test file "%s" has been generated successfully', $result->getTestPath()));
+            }
+        } catch (
+            ConfigException |
+            GeneratorException |
+            RendererException |
+            SutException $exception) {
+            throw $this->handleException($exception, $isVerbose);
+        }
+
+        return self::CODE_SUCCESS;
+    }
+
+    /**
+     * @param SourceFileCollection $sourceFileCollection
+     *
+     * @throws GeneratorException
+     * @throws MissingSourceFileException
+     */
+    private function validateSourceFileCollection(SourceFileCollection $sourceFileCollection): void
+    {
+        if (!$sourceFileCollection->hasSome()) {
+            throw new GeneratorException('No source file(s).');
+        }
+
+        if ($sourceFileCollection->hasMissing()) {
+            $exception = new MissingSourceFileException('Source files does not exist.');
+            $exception->setMissingFiles($sourceFileCollection->getMissing());
+
+            throw $exception;
+        }
+    }
+
+    /**
+     * @param InputInterface $input
+     *
+     * @return string
+     */
+    private function getConfigFile(InputInterface $input): ?string
+    {
+        /** @var string $configParam */
+        $configParam = $input->getOption(self::OPTION_CONFIG_FILE);
+        $configPath = realpath($configParam);
+
+        return $configPath === false
+            ? null
+            : $configPath;
+    }
+
+    /**
+     * @param InputInterface $input
+     *
+     * @return string[]
+     */
+    private function getSourceFiles(InputInterface $input): array
+    {
+        /** @var string[] $sourceFiles */
+        $sourceFiles = $input->getArgument(self::ARG_FILES);
+
+        return $sourceFiles;
+    }
+
+    /**
+     * @param InputInterface $input
+     *
+     * @return bool
+     */
+    private function getOverrideFlag(InputInterface $input): bool
+    {
+        /** @var bool $override */
+        $override = $input->getOption(self::OPTION_OVERRIDE_FILE);
+
+        return $override;
+    }
+
+    /**
+     * @param Throwable $exception
+     * @param bool $isVerbose
+     *
+     * @return GenerateCommandException
+     */
+    private function handleException(Throwable $exception, bool $isVerbose): GenerateCommandException
```

Comment:
In my opinion, we should consider validation before and just throw simple exceptions in runtime.